app/service/hotel_service.py

```diff
@@ -0,0 +1,47 @@
+from app.database import db
+from app.util import formulate_date, get_days_between, add_days
+
+
+def read_hotel(checkInDate: str, checkOutDate: str, destination: str):
+    checkIn_datetime = formulate_date(checkInDate)
+    checkOut_datetime = formulate_date(checkOutDate)
+    no_of_days = get_days_between(checkIn_datetime, checkOut_datetime)
+    response = []
+
+    available_hotels = list(
```

Comment:
Not too sure if you can do this, but would it be possible to structure a query where you can get all hotels from start to end date? Then you can aggregate the total amount by just looping the records, only requiring 1 query.

Right now it seems like for n records, you are doing n+1 queries, which is rather expensive. Generally, it would be good to avoid multiple calls to db if possible.

Reply:
Noted. I have removed the original implementation by using the mongoDB aggregate method and thus also removed the unnecessary methods used in date.utils